src/domain/dataset-update-flow/src/entities/update/update_event.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, Utc};
+use enum_variants::*;
+use kamu_task_system::{TaskID, TaskOutcome};
+use opendatafabric::{AccountID, AccountName, DatasetID};
+
+use crate::*;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+/// All events that model life-cycle of a task
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum UpdateEvent {
+    /// Update initiated
+    Initiated(UpdateEventInitiated),
+    /// Start condition defined
+    StartConditionDefined(UpdateEventStartConditionDefined),
```

Comment:
This is only used for derived dataset, to show why the launch is waiting in the History tab. I think it wasn't me, who suggested the term "start condition".

Reply:
The name is fine, I'm just asking if start condition can be specified during aggregate creation and appear as a field in the `Initiated` event instead of having its own event.

Reply:
The experiment failed, we have a single aggregate finally.